source/declarations.tex

```diff
@@ -4361,7 +4361,22 @@
 and the initializer is a string literal, see~\ref{dcl.init.string}.
 \item If the initializer is \tcode{()}, the object is value-initialized.
 \item
-Otherwise, if the destination type is an array, the program is ill-formed.
+Otherwise, if the destination type is an array,
+the object is initialized as follows.
+Let $x_1$, $\dotsc$, $x_k$ be
+the elements of the \grammarterm{expression-list}.
+If the destination type is an array of unknown bound,
+it is defined as having $k$ elements.
+If $k$ is greater than the size of the array,
+the program is ill-formed.
+Otherwise, the $i^\text{th}$ array element is copy-initialized with
+$x_i$ for each $1 \leq i \leq k$, and
+value-initialized for each $k < i \leq n$.
```

Comment:
We don't introduce "n" anywhere for this array case. We could say "If $k$ is greater than the size of the array $n$" a few lines prior to here.

Reply:
Maybe add:

> as having $k$ elements<ins>; either way, let $n$ be the array size of the destination type</ins>.

Reply:
I went with:
`... ; either way, let $n$ denote the array size.`
I'm not thrilled with "either way", but it's better than nothing.

Reply:
Would you prefer "in either case"?

Reply:
I think that's slightly better, but what bothers me is, what are the cases?  this is in a sentence where we've only introduced one case.  Making it a separate sentence still has a similar problem.

Reply:
Oh, right: the two cases are known and unknown bound. So you could say, "regardless of whether the original type was of known or unknown bound, let $n$ denote the array size."

Or: "Let $n$ denote the array size after this potential adjustment."